Button failing typescript validation

- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-forked-frf6p?file=/index.js:0-231)

### Steps to reproduce
Run `tsc` for typescript checking when Button imported from antd.

### What is expected?
`tsc` should pass without errors.

### What is actually happening?
`tsc` fails with output:
```
yarn tsc
yarn run v1.22.0
$/testrepo/node_modules/.bin/tsc
node_modules/antd/lib/pagination/Pagination.d.ts:3:18 - error TS2430: Interface 'import("/testrepo/node_modules/antd/lib/pagination/Pagination").PaginationProps' incorrectly extends interface 'import("rc-pagination").PaginationProps'.
  Types of property 'itemRender' are incompatible.
    Type '((page: number, type: "next" | "prev" | "page" | "jump-prev" | "jump-next", originalElement: ReactElement<HTMLElement, string | JSXElementConstructor<any>>) => ReactNode) | undefined' is not assignable to type '((page: number, type: string, element: ReactNode) => ReactNode) | undefined'.
      Type '(page: number, type: "next" | "prev" | "page" | "jump-prev" | "jump-next", originalElement: ReactElement<HTMLElement, string | JSXElementConstructor<any>>) => ReactNode' is not assignable to type '(page: number, type: string, element: ReactNode) => ReactNode'.
        Types of parameters 'type' and 'type' are incompatible.
          Type 'string' is not assignable to type '"next" | "prev" | "page" | "jump-prev" | "jump-next"'.

3 export interface PaginationProps extends RcPaginationProps {
                   ~~~~~~~~~~~~~~~


Found 1 error.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

| Environment | Info |
|---|---|
| antd | 4.17.0 |
| React | 17.0.2 |
| System | Mac |
| Browser | Chrome |


<!-- generated by ant-design-issue-helper. DO NOT REMOVE -->

Relevant closed issues: https://github.com/ant-design/ant-design/issues/32222, https://github.com/ant-design/ant-design/issues/32206

--- a/components/pagination/Pagination.tsx
+++ b/components/pagination/Pagination.tsx
@@ -20,11 +20,6 @@ export interface PaginationProps extends RcPaginationProps {
   showQuickJumper?: boolean | { goButton?: React.ReactNode };
   size?: 'default' | 'small';
   responsive?: boolean;
-  itemRender?: (
-    page: number,
-    type: 'page' | 'prev' | 'next' | 'jump-prev' | 'jump-next',
-    originalElement: React.ReactElement<HTMLElement>,
-  ) => React.ReactNode;
   role?: string;
   totalBoundaryShowSizeChanger?: number;
 }

